src/Simulations/simulation.jl

```diff
@@ -1,47 +1,61 @@
 """
     Simulation <: AbstractSimulation
 """
-struct Simulation{MT,TT,OT} <: AbstractSimulation
-    model::MT
-    timestepper::TT
-    ode_problem::OT
+struct Simulation <: AbstractSimulation
+    model::AbstractModel
+    integrator::DEIntegrator
 end
 
 """
-    Simulation(model::AbstractModel, stepper::AbstractTimestepper)
+    Simulation(model::AbstractModel, method::AbstractTimestepper; dt, tspan)
 """
-function Simulation(; model::AbstractModel, stepper::AbstractTimestepper)
-    ode_function   = make_ode_function(model)
-    state_init     = make_initial_conditions(model)
-    ode_problem    = DiffEqBase.ODEProblem(ode_function, state_init, stepper.tspan)
+function Simulation(model::AbstractModel, method; dt, tspan, kwargs...)
+    ode_function = make_ode_function(model)
+    Y = make_initial_conditions(model)
```

Comment:
This may need to be split, so that you can initiate your state externally, when/if we decide to enable multiple models' state vectors to be combined and passed around as one Y vector. But we can cross that bridge then.

Reply:
Good point. There are so many cases :D.